beast/observationmodel/ast/make_ast_input_list.py

```diff
@@ -432,34 +441,53 @@ def supplement_ast(
         print(
             "{} exists. Will attempt to load SEDs for ASTs from there \
             and remove those SEDs from the SED grid".format(
-                existingASTfile))
+                existingASTfile
+            )
+        )
 
-        t = Table.read(existingASTfile, format="fits")
-        sedsMags = np.delete(sedsMags, t["sedgrid_indx"], axis=0)
-        sedsIndx = np.delete(sedsIndx, t["sedgrid_indx"])
+        t = Table.read(existingASTfile, format="ascii")
+        sedsMags = np.delete(sedsMags, np.array(t["sedgrid_indx"], dtype=int), axis=0)
+        sedsIndx = np.delete(sedsIndx, np.array(t["sedgrid_indx"], dtype=int))
         Nseds = sedsMags.shape[0]
 
     # Apply selection conditions if supplied
+    # Just magnitude cuts
+    print("mag_cuts", mag_cuts)
+    print("color_cuts", color_cuts)
     if mag_cuts is not None:
-        cond = np.zeros(Nseds, dtype=bool)
-
+        cond = np.ones(Nseds, dtype=bool)
         for key in list(mag_cuts.keys()):
             idx_filter = [i for i, iflt in enumerate(filters) if key in iflt]
             bright_cut = mag_cuts[key][0]
             faint_cut = mag_cuts[key][1]
-            tmp_cond = np.logical_and(sedsMags[:, idx_filter] >= bright_cut,
-                                      sedsMags[:, idx_filter] <= faint_cut
-                                      )
+            tmp_cond = np.logical_and(
+                (sedsMags[:, idx_filter] >= bright_cut),
+                (sedsMags[:, idx_filter] <= faint_cut),
+            )
 
-            cond = np.logical_or(cond, tmp_cond.ravel())
+            if color_cuts is not None:
+                if key in color_cuts:
+                    for limit in color_cuts[key]:
+
+                        idx_color_filter = [
+                            i for i, iflt in enumerate(filters) if limit[0] in iflt
+                        ]
+                        tmp_cond = np.logical_and(
+                            tmp_cond,
+                            (
+                                sedsMags[:, idx_filter] - sedsMags[:, idx_color_filter]
+                                <= limit[1]
+                            ),
+                        )
+            cond = np.logical_and(cond, tmp_cond.ravel())
 
         sedsMags = sedsMags[cond, :]
         sedsIndx = sedsIndx[cond]
 
     # Randomly select models
     # Supplementing ASTs does not need to follow
     # the toothpick-way selection
-    chosen_idxs = np.random.choices(np.arange(len(sedsIndx)), k=nAST)
+    chosen_idxs = np.random.choice(np.arange(len(sedsIndx)), nAST)
```

Comment:
Oh! It seems like we can just provide an int for the first parameter, i.e., np.random.choice(len(sedsIndx), nAST).

a : 1-D array-like or int
    If an ndarray, a random sample is generated from its elements.
    If an int, the random sample is generated as if a were np.arange(a)